streamlit/data_source.py

```diff
@@ -0,0 +1,19 @@
+"""Functions that handle communication with the database."""
+
+from os import environ as ENV
+from psycopg2 import connect
+from psycopg2.extensions import connection
+import streamlit as st
+
+@st.cache_resource
+def get_db_connection() -> connection:
+    """Returns a live connection to the local PostgreSQL database."""
+    config = {
+        "dbname": ENV.get("DB_NAME"),
+        "user": ENV.get("DB_USER"),
+        "password": ENV.get("DB_PASSWORD"),
+        "host": ENV.get("DB_HOST"),
+        "port": ENV.get("DB_PORT")
+    }
+
+    return connect(**config)
```

Comment:
This is fine, but a little odd; why not just do this?
```suggestion
    conn = connect(
        "dbname": ENV.get("DB_NAME"),
        "user": ENV.get("DB_USER"),
        "password": ENV.get("DB_PASSWORD"),
        "host": ENV.get("DB_HOST"),
        "port": ENV.get("DB_PORT")
    )

    return conn
```

Reply:
It's not allowing me to run it like that.

Reply:
Changes made now!